dhall/src/Dhall/Import.hs

```diff
@@ -746,6 +749,36 @@ toHeader _ = do
 getCacheFile
     :: (Alternative m, MonadIO m) => FilePath -> Dhall.Crypto.SHA256Digest -> m FilePath
 getCacheFile cacheName hash = do
+    cacheDirectory <- getOrCreateCacheDirectory cacheName
+
+    let cacheFile = cacheDirectory </> ("1220" <> show hash)
+
+    return cacheFile
+
+warnAboutMissingCaches  :: (Alternative m, MonadIO m) => m ()
+warnAboutMissingCaches = do
+    warnings <- traverse warnAboutMissingCache ["dhall", "dhall-haskell"]
+
+    case catMaybes warnings of
+        warn:_ -> liftIO (System.IO.hPutStrLn System.IO.stderr warn)
+        []     -> return () 
+      
+
+warnAboutMissingCache :: (Alternative m, MonadIO m) => FilePath -> m (Maybe String)
+warnAboutMissingCache cacheName = alternative₀ <|> alternative₁
+  where
+    
+    alternative₀ = getOrCreateCacheDirectory cacheName >> return Nothing
+
+    alternative₁ = return (Just warning)
+
+      where warning = "It has not been possible to get a cache directory \"" ++ cacheName ++ "\" with read/write permission.\n"
+                   ++ "You can enable cache pointing the $XDG_CACHE_HOME environment variable\n"
+                   ++ "to a directory with the required permission.\n"
+                   ++ "A subdirectory named \"" ++ cacheName ++ "\" will be created inside if it isn't exist.\n"
```

Comment:
```suggestion
                   ++ "A subdirectory named \"" ++ cacheName ++ "\" will be created inside if it doesn't already exist.\n"
```

I think a user would already expect `dhall` to create directories in the cache directory – IMHO it doesn't need to be said explicitly. So maybe this sentence could simply be removed.

Reply:
My goal was users don't try to set the base cache dir like `set $XDG_CACHE_HOME=$HOME/.cache/dhall` although the env variable name suggests that it is a general dir for cache.